### DISCUSSION TITLE
适用于中国大陆的cdnjs镜像库 Cdnjs mirror repository for mainland China

### DISCUSSION THREAD

**Discussion by apscat** [cdnjs/cdnjs#14115] [General] [answered]
## 适用于中国大陆的cdnjs镜像服务，字节跳动静态资源公共库 [https://cdn.bytedance.com/](https://cdn.bytedance.com/)

底层CDN来自 阿里CDN（服务大陆的alikulun），金山CDN，白山CDN，华为CDN
阿里kulun : https://lf3-cdn-tos.bytecdntp.com/...
金山CDN : https://lf6-cdn-tos.bytecdntp.com/...
白山云CDN : https://lf9-cdn-tos.bytecdntp.com/...
华为云CDN : https://lf26-cdn-tos.bytecdntp.com/...

cdnjs mirroring service for mainland China, byte-hopping static resource public library [https://cdn.bytedance.com/](https://cdn.bytedance.com/)
The underlying CDN comes from Ali CDN (serving mainland alikulun), Jinshan CDN, Baishan CDN, Huawei CDN

**Comment by apscat** [comment]
该服务在中国大陆地区有较高的服务稳定性与较低的延迟
The service has high service stability and low latency in mainland China

**Comment by wfjsw** [comment]
There seems to be issues within this mirror. 
- Libraries have not been updated since `2021-08-30`.
- The `datatables.net` series of packages have weird versioning and invalid content.
- No contact info.
- Default expiry set to 1 month

**Reply by apscat** [reply]
I sent them an email and have solved all the problems
Now you can manually select the cache time
and the library is up to date
![1](https://ae05.alicdn.com/kf/H02c50c12775e4c8bbde225b8c586262ec.png)

**Reply by wfjsw** [reply]
It's no longer updated at all. I poked an intern at ByteDance the next day I posted that and it seems they only did a one-time update.

**Reply by apscat** [reply]
No one seems to maintain it, but cdn is really fast.

**Reply by cxw620** [reply]
> I sent them an email and have solved all the problems Now you can manually select the cache time and the library is up to date ![1](https://camo.githubusercontent.com/1539741f2b835752ee733e2635d66137a390e1be7e7df0e780c09d2b4457aace/68747470733a2f2f616530352e616c6963646e2e636f6d2f6b662f483032633530633132373735653463386262646532323562386335383632363265632e706e67)

Would you please share their email address?

**Comment by sparkcyf** [comment]
It looks like the cdnjs images that has server in Mainland China and still being regularly updated are:

- ~~https://www.bootcdn.cn/ (last updated in about 2022-04, however The CDN domain `cdn.bootcdn.net` seems to be resolved to cloudflare)~~
- https://cdn.baomitu.com/ (Hosted by 360, last updated in about 2022-03)
- https://mirrors.cqupt.edu.cn/cdnjs/ (Hosted by CQUPT, last update on 2022-03-29)
- https://mirrors.sustech.edu.cn/cdnjs/ (Hosted by SUSTech CRA, last update on 2022-05-22)

## TCP Ping result of the sites above
### `cdn.bootcdn.net`:
![image](https://user-images.githubusercontent.com/14960340/169675663-fc3c7400-4ff3-4cfe-b7cd-369634446ea7.png)

### `lib.baomitu.com`:
![image](https://user-images.githubusercontent.com/14960340/169675683-616c8b9e-9486-4129-a169-c3950f784f09.png)

### `mirrors.cqupt.edu.cn`:
![image](https://user-images.githubusercontent.com/14960340/169675687-adf9e90a-8e8a-4998-a086-5ad5080740d6.png)

### `mirrors.sustech.edu.cn`:
![image](https://user-images.githubusercontent.com/14960340/169675697-4f350169-4369-4317-b8ca-2cb4222bfb0e.png)

**Comment by lbnliboning** [comment]
OK.That’s very great.

**Comment by Francklin** [comment]
还有web缓存网https://www.webcache.cn，提供常用前端公共JS库

**Comment by LeterTW** [comment]
字节和360的都很久没有同步文件了
https://cdnjs.snrat.com/ 白山云CDN
https://www.zstatic.net/ 又拍云赞助CDN
https://www.7ed.net/start/public-cdn.html 华为云CDN

**Comment by wfjsw** [comment]
PSA: Properly set up SRI when using 3rd party CDNJS mirrors or your site may be fried in the future.

**Comment by lazychanger** [comment]
已踩坑。

注意CDN来源，请勿使用个人或未备案域名，谨防CDN投毒。

推荐：mirrors.sustech.edu.cn

**Reply by alswl** [reply]
mirrors.sustech.edu.cn 没有开启 CORS。

**Reply by lazychanger** [reply]
换别的吧，或者直接使用cdnjs.cloudflare.com

**Reply by alswl** [reply]
放弃了，准备自建了。

**Reply by lazychanger** [reply]
通过CDN回源到自己的云存储，再通过云存储进行二次回源到cdnjs.cloudflare.com，可以有效避免撤包等操作。
注意配置回源时候去掉`accept`头，避免cloudlflare回源的是gzip或者dz压缩格式文件。

**Comment by ajz34** [comment, marked as answer]
@apscat 
请留意，最近字节跳动静态资源公共库已经下线。其网站说明表示是由于黑产原因而下线。
Please note that ByteDance's public static resource library has recently been taken offline. The website states that the shutdown was due to illegal activities.
